NitroxPatcher/Patches/Persistent/Language_LoadLanguageFile_Patch.cs

```diff
@@ -10,50 +12,79 @@ namespace NitroxPatcher.Patches.Persistent;
+        }
 
-        foreach (string file in files)
+        if (TryLoadLanguageFile(languageToIsoCode[language].Item2, ___strings))
         {
-            if (!File.Exists(file))
-            {
-                Log.Warn($"No language file was found for at {file}. Using English as fallback");
-                continue;
-            }
+            return;
+        }
 
-            JsonData json;
-            using (StreamReader streamReader = new(file))
-            {
-                try
-                {
-                    json = JsonMapper.ToObject(streamReader);
-                }
-                catch (Exception ex)
-                {
-                    Log.Error(ex, $"Error while reading language file {language}.json");
-                    return;
-                }
-            }
+        Log.Warn($"No language file was found for {language}. Using English as fallback");
+        TryLoadLanguageFile("en", ___strings);
+
+        Language.main.ParseMetaData();
+    }
+
+    private static bool TryLoadLanguageFile(string fileName, IDictionary<string, string> strings)
+    {
+        string filePath = Path.Combine(NitroxUser.LauncherPath, "LanguageFiles", $"{fileName}.json");
+
+        if (!File.Exists(filePath))
+        {
+            return false;
+        }
+
+        using StreamReader streamReader = new(filePath);
+        try
+        {
+            JsonData json = JsonMapper.ToObject(streamReader);
 
             foreach (string key in json.Keys)
             {
-                if (json[key].IsString)
+                JsonData jsonKey = json[key];
+                if (jsonKey.IsString)
                 {
-                    ___strings[key] = (string)json[key];
+                    strings[key] = (string)jsonKey;
                 }
             }
+
+            return true;
+        }
+        catch (Exception ex)
+        {
+            Log.Error(ex, $"Error while reading language file {fileName}.json");
         }
 
+        return false;
     }
 
     public override void Patch(Harmony harmony)
     {
-        PatchPostfix(harmony, TARGET_METHOD);
+        List<string> existingLanguageNames = Directory.EnumerateFiles(SNUtils.InsideUnmanaged("LanguageFiles"), "*.json").Select(Path.GetFileNameWithoutExtension).ToList();
+
+        foreach (CultureInfo culture in CultureInfo.GetCultures(CultureTypes.AllCultures))
+        {
+            if (!languageToIsoCode.ContainsKey(culture.EnglishName) && !string.IsNullOrEmpty(culture.Name) && existingLanguageNames.Contains(culture.EnglishName))
+            {
+                languageToIsoCode.Add(culture.EnglishName, new Tuple<string, string>(culture.Name, culture.TwoLetterISOLanguageName));
+            }
+        }
+
+        // This language isn't registered in CultureInfo
+        if (existingLanguageNames.Contains("Spanish (Latin America)"))
+        {
+            languageToIsoCode.Add("Spanish (Latin America)", new Tuple<string, string>("es-419", "es"));
+        }
```

Comment:
Even if european spanish and latin america spanish does use different words for the same thing, I don't think it's worth doing special treatment for this one since both countries will understand the sentences (similar for US and UK english)

Reply:
While I agree with the thought that it's normally not worth I think we should support any language SN also supports, even if it's just a en vs en-UK thing.